tests/functional-tests/scale_operator.py

```diff
@@ -486,17 +489,22 @@ def test_dynamic(self, value_sc, value_vs_class=None, number_of_snapshots=None):
                 snapshot.create_vs(vs_name+"-"+str(num), vs_class_name, pvc_name)
                 snapshot.check_vs_detail(vs_name+"-"+str(num), pvc_name, vs_class_name, sc_name, value_vs_class)
             
-            for num in range(0, number_of_restore): 
-                restored_pvc_name = "restored-pvc"+vs_name[2:]+"-"+str(num)
-                snap_pod_name = "snap-end-pod"+vs_name[2:]
-                d.create_pvc_from_snapshot(pvc_value, sc_name, restored_pvc_name,vs_name+"-"+str(num))
-                d.check_pvc(pvc_value, sc_name, restored_pvc_name)
-                d.create_pod(value_pod, restored_pvc_name, snap_pod_name,self.image_name)
-                d.check_pod(value_pod, sc_name, restored_pvc_name, snap_pod_name)
-                d.check_file_inside_pod(value_pod, sc_name, restored_pvc_name, snap_pod_name)
-                d.delete_pod(snap_pod_name)
-                d.check_pod_deleted(snap_pod_name)
-                d.delete_pvc(restored_pvc_name)
+            if not(ff.snapshot_restore_available()): 
+                pvc_value["reason"] = "Min required Spectrum Scale version is 5.0.5.2"
+
+            if test_restore:
+                for num in range(0, number_of_restore): 
+                    restored_pvc_name = "restored-pvc"+vs_name[2:]+"-"+str(num)
+                    snap_pod_name = "snap-end-pod"+vs_name[2:]
+                    d.create_pvc_from_snapshot(pvc_value, sc_name, restored_pvc_name,vs_name+"-"+str(num))
+                    val = d.check_pvc(pvc_value, sc_name, restored_pvc_name)
+                    if val is True:
+                        d.create_pod(value_pod, restored_pvc_name, snap_pod_name,self.image_name)
+                        d.check_pod(value_pod, sc_name, restored_pvc_name, snap_pod_name)
+                        d.check_file_inside_pod(value_pod, sc_name, restored_pvc_name, snap_pod_name)

```

Comment:
Are we also checking if the restored PVC has all the data from snapshot?

Reply:
Yes, we are checking whether the written file from original pod and in snapshot is there on the restore pvc.